File: 1/Memory.java
public class Memory extends Hardware{
	// Atribut
	private int frequency;			// Frequency
	private int memorySize;			// Memory Size
	private boolean supportsCuda;	// Cuda

	// Konstruktor
	public Memory() {
		frequency = 0;
		memorySize = 0;
		supportsCuda = true;
	}

	/* Set & Get */

	// Frequency
	public void setFrequency(int frequency) {
		this.frequency = frequency;
	}
	public int getFrequency() {
		return this.frequency;
	}

	// Memory Size
	public void setMemorySize(int memorySize) {
		this.memorySize = memorySize;
	}
	public int getMemorySize() {
		return this.memorySize;
	}

	// Cuda
	public void setSupportsCuda(char supportsCuda) {
		this.supportsCuda = (supportsCuda == 'y') ? true : false;
	}
	public boolean getSupportsCuda() {
		return this.supportsCuda;
	}
}

File: 2/C++/Memory.cpp
class Memory: public Hardware {
	private:
		// Atribut
		int frequency;
		int memorySize;
		bool supportsCuda;

	public:
		// Kosntruktor
		Memory() {
			frequency = 0;
			memorySize = 0;
			supportsCuda = true;
		}

		// Set & Get

		// Frequency
		void setFrequency(int frequency) {
			this->frequency = frequency;
		}
		int getFrequency() {
			return frequency;
		}

		// Memory Size
		void setMemorySize(int memorySize) {
			this->memorySize = memorySize;
		}
		int getMemorySize() {
			return memorySize;
		}

		// Cuda
		void setSupportsCuda(char supportsCuda) {
			this->supportsCuda = (supportsCuda == 'y') ? true : false;
		}
		int getSupportsCuda() {
			return supportsCuda;
		}

		// Destruktor
		~Memory(){}
};